twitter/views.py

```diff
@@ -78,62 +77,62 @@ def signup(request):
-#             context['connected'] = True if result else False
-
-#         return context
-
-# @login_required
-# def follow_view(request, *args, **kwargs):
-#     try:
-#         follower = User.objects.get(username=request.user)
-#         following = User.objects.get(username=kwargs['username'])
-#     except User.DoesNotExist:
-#         messages.warning(request, '{}は存在しません'.format(kwargs['username']))
-#         return HttpResponseRedirect(reverse_lazy('twitter:index'))
-
-#     if follower == following:
-#         messages.warning(request, '自分自身はフォローできませんよ')
-#     else:
-#         _, created = Connection.objects.get_or_create(follower=follower, following=following)
-
-#         if (created):
-#             messages.success(request, '{}をフォローしました'.format(following.username))
-#         else:
-#             messages.warning(request, 'あなたはすでに{}をフォローしています'.format(following.username))
-
-#     return HttpResponseRedirect(reverse_lazy('twitter:profile', kwargs={'username': following.username}))
-
-# @login_required
-# def unfollow_view(request, *args, **kwargs):
-#     try:
-#         follower = User.objects.get(username=request.user)
-#         following = User.objects.get(username=kwargs['username'])
-#         if follower == following:
-#             messages.warning(request, '自分自身のフォローを外せません')
-#         else:
-#             unfollow = Connection.objects.get(follower=follower, following=following)
-#             unfollow.delete()
-#             messages.success(request, 'あなたは{}のフォローを外しました'.format(following.username))
-#     except User.DoesNotExist:
-#         messages.warning(request, '{}は存在しません'.format(kwargs['username']))
-#         return HttpResponseRedirect(reverse_lazy('twitter:index'))
-#     except Connection.DoesNotExist:
-#         messages.warning(request, 'あなたは{0}をフォローしませんでした'.format(following.username))
-
-#     return HttpResponseRedirect(reverse_lazy('twitter:profile', kwargs={'username': following.username}))
\ No newline at end of file
+class ProfileDetailView(LoginRequiredMixin, DetailView):
+     model = User
+     template_name = "twitter/profile.html"
+     slug_field = 'username'
+     slug_url_kwarg = 'username'
+
+     def get_context_data(self, **kwargs):
+         context = super(ProfileDetailView, self).get_context_data(**kwargs)
+         username = self.kwargs['username']
+         context['username'] = username
+         context['user'] = self.request.user
+         context['following_count'] = Connection.objects.filter(follower__username=username).count()
+         context['followers_count'] = Connection.objects.filter(following__username=username).count()
+
+         if username != context['user'].username:
+             result = Connection.objects.filter(follower__username=context['user'].username, following__username=username)
+             context['connected'] = True if Connection.objects.exists() else False
```

Comment:
result変数が定義されていますが、未使用の状態になっているのでfilterの条件について変化がありません。こちら対応をお願いします！
また、過去にコメントした内容の補足になりますが、`exists()`はbool値を返すので最終的に以下のように一文で記述することが可能になると思うので検証してみてください。
```suggestion
             context['connected'] = Connection.objects.filter(follower__username=context['user'].username, following__username=username).exists()
```

> result(すなわちQuerySet)が存在するかどうかで真偽値を代入していますが、QuerySetAPIの`exists()`メソッドを使用するとより簡潔に書けると思うので以下のドキュメントを読んでみてください。
> docs: https://docs.djangoproject.com/en/4.0/ref/models/querysets/#django.db.models.query.QuerySet.exists